### ISSUE TITLE
Best practices on repeatedly generating experience and training on it?

### ISSUE THREAD

**Issue by paulbricman** [CarperAI/trlx#83]
All the current (online RL) trlx examples seem to only involve generating experience once and then using the resulting rollout store to update the weights (for potentially more epochs). How should one go about incrementally generating experience, training on it, generating XP w/ updated model, training again...? I thought of just calling `orch.make_experience()` and `model.learn()` in a loop multiple times, but that sounds pretty dumb. Is there a better way?

Loosely related, from "Fine-Tuning Language Models from Human Preferences":
>  If the trained policy π is very different from the zero-shot
> policy ρ, the reward model will suffer a large distributional
> shift from training on samples from ρ to evaluation on sam-
> ples from π. To prevent this, we can collect human data
> throughout RL fine-tuning, continuously gathering new data
> by sampling from π and retraining the reward model. As
> section 3 shows, online data collection was important for
> summarization but not for the simpler style tasks.

Question can also be thought of as the one cycle in the computational-ish graph in the section at the bottom from `trl`:
![image](https://user-images.githubusercontent.com/20104026/200852431-9cb973da-d11c-49c5-93bb-daa058cd23ec.png)

**Comment by paulbricman** [comment]
`trlx.model.accelerate_ppo_model` contains:

```python
self.n_updates_per_batch = self.config.method.ppo_epochs
        self.total_steps = (
            self.config.train.epochs
            * self.n_updates_per_batch
            * len(self.train_dataloader)
        )
        self.total_steps = min(self.total_steps, self.config.train.total_steps)
```

However, `ppo_epochs` seems to code for the number of gradient update steps per batch, not really for how many times to generate different experiences and train on them incrementally. Though after first seeing the `ppo_epochs` key in configs I thought that was what it was. I would rename `ppo_epochs` to something like `n_updates_per_batch` and instead reserve `ppo_epochs` for this online RL param.

**Comment by paulbricman** [comment]
Oh, actually this is already implemented. In https://github.com/CarperAI/trlx/blob/master/trlx/model/accelerate_ppo_model.py#L106:

```python
    def post_epoch_callback(self):
        self.store.clear_history()
        self.orch.make_experience(
            self.config.method.num_rollouts, self.iter_count
        )  # Collect more rollouts for training
```

Called in https://github.com/CarperAI/trlx/blob/8a11ae05cdbe1fd7e18295d03b7c7047b064111b/trlx/model/accelerate_base_model.py#L258
